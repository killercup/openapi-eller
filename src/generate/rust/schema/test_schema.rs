const SIMPLE_SCHEMA: &str = "
openapi: 3.0.1
info: { title: Foo, version: 0.1.0 }
paths: {}
components:
    schemas:
        Foo:
            type: object
            properties:
                bar:
                    type: string
";

#[test]
fn simple_schema() {
    let api: openapiv3::OpenAPI = serde_yaml::from_str(SIMPLE_SCHEMA).unwrap();

    insta::assert_display_snapshot_matches!(
        "simple_schema",
        crate::generate::rust::types(&api).unwrap()
    );
}

const NESTED_SCHEMA: &str = "
openapi: 3.0.1
info: { title: Foo, version: 0.1.0 }
paths: {}
components:
    schemas:
        Foo:
            type: object
            properties:
                bar:
                    type: object
                    properties:
                        baz:
                            type: string
";

#[test]
fn nested_schema() {
    let api: openapiv3::OpenAPI = serde_yaml::from_str(NESTED_SCHEMA).unwrap();

    insta::assert_display_snapshot_matches!(
        "nested_schema",
        crate::generate::rust::types(&api).unwrap()
    );
}

const STRING_ENUM: &str = "
openapi: 3.0.1
info: { title: Foo, version: 0.1.0 }
paths: {}
components:
    schemas:
        Foo:
            type: object
            properties:
                bar:
                    type: string
                    enum:
                        - lorem
                        - ipsum
                        - dolor
";

#[test]
fn string_enum() {
    let api: openapiv3::OpenAPI = serde_yaml::from_str(STRING_ENUM).unwrap();

    insta::assert_display_snapshot_matches!(
        "string_enum",
        crate::generate::rust::types(&api).unwrap()
    );
}

const SCHEMA_REF: &str = "
openapi: 3.0.1
info: { title: Foo, version: 0.1.0 }
paths: {}
components:
    schemas:
        Foo:
            type: object
            properties:
                bar:
                    $ref: '#/components/schemas/Bar'
        Bar:
            type: object
            properties:
                baz:
                    type: integer
";

// FIXME: Ref schema name spaces!
#[test]
fn schema_ref() {
    let api: openapiv3::OpenAPI = serde_yaml::from_str(SCHEMA_REF).unwrap();

    insta::assert_display_snapshot_matches!(
        "schema_ref",
        crate::generate::rust::types(&api).unwrap()
    );
}
