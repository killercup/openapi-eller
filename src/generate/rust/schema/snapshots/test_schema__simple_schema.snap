---
created: "2019-06-04T10:57:07.173593Z"
creator: insta@0.8.1
source: src/generate/rust/schema/test_schema.rs
expression: "crate::generate::rust::types(&api).unwrap()"
---
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename = "Foo")]
pub struct Foo {
    #[serde(rename = "bar")]
    bar: Option<String>,
}

